code/game/objects/items/stacks/tape.dm

```diff
@@ -55,27 +56,29 @@
 	user.visible_message(span_notice("[user] begins wrapping [target] with [src]."), span_notice("You begin wrapping [target] with [src]."))
 	playsound(user, 'sound/items/duct_tape/duct_tape_rip.ogg', 50, TRUE)
 
-	if(do_after(user, 3 SECONDS, target=target))
-		playsound(user, 'sound/items/duct_tape/duct_tape_snap.ogg', 50, TRUE)
-		use(1)
-		if(istype(target, /obj/item/clothing/gloves/fingerless))
-			var/obj/item/clothing/gloves/tackler/offbrand/O = new /obj/item/clothing/gloves/tackler/offbrand
-			to_chat(user, span_notice("You turn [target] into [O] with [src]."))
-			QDEL_NULL(target)
-			user.put_in_hands(O)
-			return ITEM_INTERACT_SUCCESS
+	if(!do_after(user, 3 SECONDS, target=target))
+		return ITEM_INTERACT_BLOCKING
+
+	playsound(user, 'sound/items/duct_tape/duct_tape_snap.ogg', 50, TRUE)
+	use(1)
+	if(istype(target, /obj/item/clothing/gloves/fingerless))
+		var/obj/item/clothing/gloves/tackler/offbrand/O = new /obj/item/clothing/gloves/tackler/offbrand
+		to_chat(user, span_notice("You turn [target] into [O] with [src]."))
+		QDEL_NULL(target)
+		user.put_in_hands(O)
+		return ITEM_INTERACT_SUCCESS
 
-		if(target.get_embed() && target.get_embed().type == conferred_embed)
-			to_chat(user, span_warning("[target] is already coated in [src]!"))
-			return ITEM_INTERACT_BLOCKING
+	if(target.get_embed()?.type == conferred_embed)
```

Comment:
`istype(target.get_embed(), conferred_embed)`? Or does it rely on the typecheck exactly

Reply:
relies on exact check because subtypes of tape use subtypes of embed data